### DISCUSSION TITLE
Schedule extender hiddendays update

### DISCUSSION THREAD

**Discussion by amittel** [primefaces/community#1256] [PrimeFaces] [answered]
Hello,
my plan is to dynamicllay show the weekday saturday. I tried to implement it with the schedule extender and the hiddendays option and a boolean checkbox.

Setting the string in my bean as `[0,6]` (hide sunday, saturday) and `[0]` (hide sunday).

My problem I have is, that my state of the checkbox corresponds to the String of `hiddenDays`, but the schedule is one state behind.
Example: I tick the saturday checkbox, hiddendays is updated correctly. But schedule is only showing Mo-Fr. If I untick the checkbox, the schedule displays Mo-Sa. But that should be the state when I tick the checkbox.

My frontend has a `selectBooleanCheckbox ` as following
```
<p:selectBooleanCheckbox 
            value="#{testView.saturday}" 
            itemLabel="Saturday">
            <p:ajax 
                listener="#{testView.onSaturdayChanged}"
                update="scheduleExtender,schedule00, hDays"/>
        </p:selectBooleanCheckbox>
```
`saturday` is a `boolean`
and
`onSaturdayChanged` is implemented as
```
public void onSaturdayChanged() {   
        if(saturday) {
            hiddenDays = "0";    
        }
        else {
            hiddenDays = "0,6";     
        }  
    }
```
´hiddenDays` is just a `String` which gets inserted into the javascript as
```
<p:panel id="scheduleExtender">
            <script >
                function initSchedule() {
                    $.extend(this.cfg.options, {
                    hiddenDays: [#{testView.hiddenDays}],
                    });
                }
            </script>
            </p:panel>

```

I have found a workaround, but I am still not sure why the other way is not working.
Workaround:
* Set hiddendays static to 0 (hide sunday)
* Work with attribute `showWeekends` with my `boolean saturday`

Reproducer example: [schedule-extender.zip](https://github.com/primefaces/community/files/14372257/schedule-extender.zip)

**Comment by melloware** [comment]
I have a feeling it is because of your `update="scheduleExtender,schedule00, hDays"`.  You are presupposing the DOM of the JS Extender gets replaced and then the schedule runs iwth that new updated initSchedule() function. Most likely the Schedule initializes with the current extender on update and then the Script is being replaced after.  Just a guess but that would explain your behavior....

**Reply by amittel** [reply]
Yes, looks like it. 
The [Schedule Extender showcase](https://www.primefaces.org/showcase/ui/data/schedule/extender.xhtml) has the same issue, if you select _Business hours_. It should switch to Monday to Thursday, but stays at the whole week.

**Reply by melloware** [reply, marked as answer]
One workaround you can try is this...

1. Only update `scheduleExtender` in your checkbox.
2. In your schedule extender script add a 1 line call to a remote Command.

```xml
<script >
                function initSchedule() {
                    $.extend(this.cfg.options, {
                    hiddenDays: [#{testView.hiddenDays}],
                    });
                }

               // call remote command
               rcUpdateSchedule();
            </script>
```

3. Have a remote command that does this...

```xml
<p:remoteCommand name="rcUpdateSchedule" update="schedule00, hDays" />
```

Basically this remote command will update the `schedule00, hDays` only AFTER the script has been updated in the DOM.
